### PR TITLE
clean up group generation; output to XML

### PR DESCRIPTION
**Pull request by malaffoon**
Once this is checked in we can decide if we want to fake groups in the exam-processor.

### CODE REVIEW

File: data_generator/sbac_generators/hierarchy.py [hunk 1]
```diff
@@ -189,12 +191,11 @@ def populate_schools_with_groupings(schools_with_groupings, id_gen):
     """
 
     for school, grades in schools_with_groupings.items():
-        num_groups = school.student_count_avg / sbac_config.STUDENTS_PER_GROUP
```
Comment (malaffoon) [inline data_generator/sbac_generators/hierarchy.py]: Before this was not creating groups for any school with less than 50 students.

File: data_generator/model/school.py [hunk 2]
```diff
@@ -43,3 +44,15 @@ def student_count_avg(self):
...
+    def group_size(self):
+        """The average group size
+        """
+        return self.config['group_size']
+
+    @property
+    def num_groups(self):
+        """The number of groups in a school (~ student count / group size)
+        """
+        return ceil(self.student_count_avg / self.group_size)
```
Comment (malaffoon) [inline data_generator/model/school.py]: Number of groups for a school should depend on the size of the school. It should always be at least one.

File: data_generator/sbac_generators/hierarchy.py [hunk 3]
```diff
@@ -213,12 +214,13 @@ def set_up_schools_with_groupings(schools, grades_of_concern):
         grade_sub_groups = {}
         grades_for_school = grades_of_concern.intersection(school.config['grades'])
 
+        group_types = list(sbac_config.GROUP_TYPE)
+        if random.random() >= sbac_config.STAFF_GROUP_RATE: group_types.remove('staff_based')
```
Comment (malaffoon) [inline data_generator/sbac_generators/hierarchy.py]: 'section_based' groups are meant to represent classes/sessions. 'staff_based' groups i'm leaving to be particular staff member groups; and i'm making them less common.

File: data_generator/sbac_generators/population.py [hunk 4]
```diff
@@ -146,54 +143,6 @@ def assign_student_groups(school, grade, grade_students, schools_with_groupings)
             setattr(grade_student, "group_%i_id" % (n,), group.id)
             setattr(grade_student, "group_%i_text" % (n,), group.name)
 
-            # for n in range(1, 11):
```
Comment (malaffoon) [inline data_generator/sbac_generators/population.py]: just removing lava code

File: data_generator/config/hierarchy.py [hunk 5]
```diff
@@ -45,88 +45,104 @@
     'High School':
         {'type': 'High School',
          'grades': [11],
-         'students': {'min': 300, 'max': 400, 'avg': 350}
+         'students': {'min': 300, 'max': 400, 'avg': 350},
+         'group_size': 40
```
Comment (agorina) [inline data_generator/config/hierarchy.py]: is this a percent of the 'students'?

File: data_generator/model/school.py (hunk 2 above)
Comment (agorina) [inline data_generator/model/school.py]: Oh, never mind my question above.

**Pull request merged**